File: java/javafind/lib/src/main/java/javafind/FindError.java
package javafind;

public enum FindError {
    STARTPATH_NOT_DEFINED("Startpath not defined"),
    STARTPATH_NOT_FOUND("Startpath not found"),
    STARTPATH_NOT_READABLE("Startpath not readable"),
    INVALID_RANGE_FOR_MINDEPTH_AND_MAXDEPTH("Invalid range for mindepth and maxdepth"),
    INVALID_RANGE_FOR_MINLASTMOD_AND_MAXLASTMOD("Invalid range for minlastmod and maxlastmod"),
    INVALID_RANGE_FOR_MINSIZE_AND_MAXSIZE("Invalid range for minsize and maxsize"),
    STARTPATH_DOES_NOT_MATCH_FIND_SETTINGS("Startpath does not match find settings");

    private final String message;

    FindError(final String message) {
        this.message = message;
    }

    String getMessage() {
        return message;
    }
}

File: php/phpfind/src/phpfind/FindError.php
<?php

declare(strict_types=1);

namespace phpfind;

/**
 * Enum FindError
 */
enum FindError: string
{
   case STARTPATH_NOT_DEFINED = 'Startpath not defined';
   case STARTPATH_NOT_READABLE = 'Startpath not readable';
   case STARTPATH_NOT_FOUND = 'Startpath not found';
   case INVALID_RANGE_FOR_MINDEPTH_AND_MAXDEPTH = 'Invalid range for mindepth and maxdepth';
   case INVALID_RANGE_FOR_MINLASTMOD_AND_MAXLASTMOD = 'Invalid range for minlastmod and maxlastmod';
   case INVALID_RANGE_FOR_MINSIZE_AND_MAXSIZE = 'Invalid range for minsize and maxsize';
   case STARTPATH_DOES_NOT_MATCH_FIND_SETTINGS = 'Startpath does not match find settings';
}
